com.unity.multiplayer.mlapi/Tests/Runtime/ConnectionApproval.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Text;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+using NUnit.Framework;
+
+namespace MLAPI.RuntimeTests
+{
+
+    public class ConnectionApprovalTests
+    {
+
+        private Guid m_ValidationToken;
+
+        private bool m_IsValidated;
+
+        [SetUp]
+        public void Setup()
+        {
+            // Create, instantiate, and host
+            Assert.IsTrue(NetworkManagerHelper.StartNetworkManager(out _,NetworkManagerHelper.NetworkManagerOperatingMode.None));
+            m_ValidationToken = Guid.NewGuid();
+        }
+
+        [UnityTest]
+        public IEnumerator ConnectionApproval()
+        {
+            NetworkManagerHelper.NetworkManagerObject.ConnectionApprovalCallback += NetworkManagerObject_ConnectionApprovalCallback;
+            NetworkManagerHelper.NetworkManagerObject.NetworkConfig.ConnectionApproval = true;
+            NetworkManagerHelper.NetworkManagerObject.NetworkConfig.PlayerPrefab = null;
+            NetworkManagerHelper.NetworkManagerObject.NetworkConfig.ConnectionData = Encoding.UTF8.GetBytes(m_ValidationToken.ToString());
+            m_IsValidated = false;
+            NetworkManagerHelper.NetworkManagerObject.StartHost();
+
+            var timeOut = Time.realtimeSinceStartup + 3.0f;
```

Comment:
I assume these 3.0 & 0.01 time values have a rationale to them and are compatible with how things operate on Yamato?